Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/storage/PresenceTracingRiskDaos.kt

```diff
@@ -0,0 +1,146 @@
+package de.rki.coronawarnapp.presencetracing.risk.storage
+
+import androidx.room.ColumnInfo
+import androidx.room.Dao
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.PrimaryKey
+import androidx.room.Query
+import androidx.room.TypeConverter
+import de.rki.coronawarnapp.presencetracing.risk.PtRiskLevelResult
+import de.rki.coronawarnapp.presencetracing.risk.TraceLocationCheckInRisk
+import de.rki.coronawarnapp.presencetracing.risk.calculation.CheckInWarningOverlap
+import de.rki.coronawarnapp.presencetracing.risk.calculation.PresenceTracingDayRisk
+import de.rki.coronawarnapp.presencetracing.storage.entity.TraceLocationCheckInEntity
+import de.rki.coronawarnapp.risk.RiskState
+import kotlinx.coroutines.flow.Flow
+import org.joda.time.Instant
+
+/*
+* Stores matches from the last successful execution
+* */
+@Dao
+interface TraceTimeIntervalMatchDao {
+
+    @Query("SELECT * FROM TraceTimeIntervalMatchEntity")
+    fun allMatches(): Flow<List<TraceTimeIntervalMatchEntity>>
+
+    @Query("DELETE FROM TraceTimeIntervalMatchEntity")
+    suspend fun deleteAll()
+
+    @Query("DELETE FROM TraceTimeIntervalMatchEntity WHERE endTimeMillis < :endTimeMillis")
+    suspend fun deleteOlderThan(endTimeMillis: Long)
+
+    @Query("DELETE FROM TraceTimeIntervalMatchEntity WHERE traceWarningPackageId = :warningPackageId")
+    suspend fun deleteMatchesForPackage(warningPackageId: String)
+
+    @Insert
+    suspend fun insert(entities: List<TraceTimeIntervalMatchEntity>)
+}
+
+@Entity
+data class TraceTimeIntervalMatchEntity(
+    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") val id: Long? = null,
+    @ForeignKey(
+        entity = TraceLocationCheckInEntity::class,
+        parentColumns = ["id"],
+        childColumns = ["checkInId"],
+        onDelete = ForeignKey.CASCADE
+    )
```

Comment:
As we are touching this anyway , can we change who the `ForeignKey` is referenced here? it would help upgrade to the latest version easily

Reply:
How?

Reply:
``` kotlin
@Entity(foreignKeys = [ForeignKey(entity = ParentClass::class,
                    parentColumns = arrayOf("id"),
                    childColumns = arrayOf("checkInId"),
                    onDelete = ForeignKey.CASCADE))]
```

Reply:
Changing this creates problems, so I would won't change it in this PR (I only moved the code here)